diff -urNp -x '*.orig' rt-4.4.4.org/config.layout rt-4.4.4/config.layout
--- rt-4.4.4.org/config.layout	2019-03-05 16:06:02.000000000 +0100
+++ rt-4.4.4/config.layout	2021-02-25 20:12:13.922593586 +0100
@@ -102,6 +102,37 @@
   customplugindir:  ${customdir}/plugins
 </Layout>
 
+<Layout PLDFHS>
+  prefix:		/usr/local
+  exec_prefix:		${prefix}
+  bindir:		${prefix}/bin
+  sbindir:		${prefix}/sbin
+  sysconfdir:		/etc/rt
+  datadir:		${prefix}/share
+# FIXME: missing support for lib64
+  libdir:		${prefix}/lib
+  mandir:		${datadir}/man
+# FIXME: no such directory in FHS; shouldn't go to somewhere in "${datadir}/rt/"?
+  plugindir:		${datadir}/plugins
+  htmldir:		${datadir}/rt/html
+  lexdir:		${datadir}/rt/po
+  staticdir:		${datadir}/rt/static
+  fontdir:		${datadir}/fonts/TTF
+  manualdir:		${datadir}/doc
+  localstatedir:	/var
+  logfiledir:		${localstatedir}/log
+# XXX: "/var/cache/mason/*"?
+  masonstatedir:	${localstatedir}/cache/mason_data
+  sessionstatedir:	${localstatedir}/cache/session_data
+  customdir:		${prefix}/local
+  custometcdir:	${customdir}/etc
+  customhtmldir:	${customdir}/html
+  customlexdir:	${customdir}/po
+  customstaticdir:	${customdir}/static
+  customlibdir:	${customdir}/lib
+  customplugindir:	${customdir}/plugins
+</Layout>
+
 <Layout FreeBSD>
   prefix:		/usr/local
   exec_prefix:		${prefix}
