--- rt-5.0.2/share/html/Ticket/Update.html~	2021-09-14 23:42:01.000000000 +0200
+++ rt-5.0.2/share/html/Ticket/Update.html	2022-01-31 12:57:47.459829737 +0100
@@ -410,6 +410,15 @@ if ( $ARGS{'SubmitTicket'} ) {
     }
 }
 
+if ( $ARGS{UpdateTimeWorked} ) {
+    $ARGS{UpdateTimeWorked} =~ s!^\s+!!;
+    $ARGS{UpdateTimeWorked} =~ s!\s+$!!;
+    if ( $ARGS{UpdateTimeWorked} !~ /^(?:\d+)?(?:\.(?:\d+)?)?$/ ) {
+        push @results, loc('Invalid worked time, should be numeric: "[_1]"', $ARGS{UpdateTimeWorked});
+        $checks_failure = 1;
+    }
+}
+
 # check email addresses for RT's
 {
     foreach my $field ( qw(UpdateCc UpdateBcc) ) {
diff --git a/t/web/ticket_timeworked.t b/t/web/ticket_timeworked.t
index dc52cebd84..16ad0f03a9 100644
--- a/t/web/ticket_timeworked.t
+++ b/t/web/ticket_timeworked.t
@@ -146,4 +146,70 @@ diag "checking child ticket 2 for expected timeworked data"; {
     );
 }
 
+diag "checking invalid and edge cased worked time on update";
+{
+    my $ticket = RT::Test->create_ticket( Queue => 'General', Subject => "timeworked", TimeWorked => 30, );
+    my $time = $ticket->TimeWorked;
+    is( $time, 30, "current TimeWorked" );
+
+    $m->goto_ticket( $ticket->id, 'Update' );
+    $m->submit_form_ok(
+        { form_name => 'TicketUpdate', fields => { UpdateTimeWorked => '0,1', }, button => 'SubmitTicket', },
+        'update TimeWorked with "0,1"' );
+    $m->text_contains( 'Invalid worked time, should be numeric: "0,1"' );
+
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 30, 'TimeWorked is not updated' );
+
+
+    $m->submit_form_ok(
+        {
+            form_name => 'TicketUpdate',
+            fields    => { UpdateTimeWorked => '1.5', 'UpdateTimeWorked-TimeUnits' => 'hours' },
+            button    => 'SubmitTicket',
+        },
+        'update TimeWorked with 1.5 hours'
+    );
+    $m->text_lacks( 'Invalid worked time, should be numeric' );
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 120, 'TimeWorked is updated' );
+
+    $m->goto_ticket( $ticket->id, 'Update' );
+    $m->submit_form_ok(
+        {
+            form_name => 'TicketUpdate',
+            fields    => { UpdateTimeWorked => '.5', 'UpdateTimeWorked-TimeUnits' => 'hours' },
+            button    => 'SubmitTicket',
+        },
+        'update TimeWorked with .5 hours'
+    );
+    $m->text_lacks( 'Invalid worked time, should be numeric' );
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 150, 'TimeWorked is updated' );
+
+    $m->goto_ticket( $ticket->id, 'Update' );
+    $m->submit_form_ok(
+        { form_name => 'TicketUpdate', fields => { UpdateTimeWorked => '5.' }, button => 'SubmitTicket', },
+        'update TimeWorked 5. minutes' );
+    $m->text_lacks( 'Invalid worked time, should be numeric' );
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 155, 'TimeWorked is updated' );
+
+    $m->goto_ticket( $ticket->id, 'Update' );
+    $m->submit_form_ok(
+        { form_name => 'TicketUpdate', fields => { UpdateTimeWorked => ' 15 ' }, button => 'SubmitTicket', },
+        'update TimeWorked with " 15 "' );
+    $m->text_lacks( 'Invalid worked time, should be numeric' );
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 170, 'TimeWorked is updated' );
+
+    $m->goto_ticket( $ticket->id, 'Update' );
+    $m->submit_form_ok(
+        { form_name => 'TicketUpdate', fields => { UpdateTimeWorked => ' ' }, button => 'SubmitTicket', },
+        'update TimeWorked with just a space' );
+    $m->text_lacks( 'Invalid worked time, should be numeric' );
+    $ticket->Load( $ticket->id );
+    is( $ticket->TimeWorked, 170, 'TimeWorked is not updated' );
+}
+
 done_testing();
